scipy/_lib/_util.py

```diff
@@ -28,7 +28,7 @@
         DTypePromotionError
     )
 else:
```

Comment:
```suggestion
else:  # remove this when minimum supported NumPy >= '1.25.0'
```
@lucascolley I think this is implied by the `if` condition itself, but here it is. Next time, feel free to add.

Reply:
oops, I didn't look carefully enough at the `if` condition 🤦‍♂️ sorry for the noise!